PostHog/Replay/PostHogSessionReplayConfig.swift

```diff
@@ -50,6 +50,18 @@
         /// Defaults to 1s
         @objc public var debouncerDelay: TimeInterval = 1.0
 
+        /// Whether to start session replay automatically or manually during the SDK setup
+        /// Options are:
+        /// - .automatic: Start session replay automatically during the SDK setup
+        /// - .manual: Start/Stop session replay manually by calling `startSessionReplay()` and `stopSessionReplay()`
+        /// Default is .automatic
+        @objc public var startMode: PostHogSessionReplayStartMode = .automatic
```

Comment:
The JS SDK does not have this since it is automatic by default, but if `sessionReplay` is disabled, they can just call `startSessionRecording`
Any reason to be different?

Reply:
> Could be a bit awkward though switching the config.sessionReplay behind the scenes for the user?

we won't do this, the user chooses to `sessionReplay: true` for everything automatically, or keep it `false` and start/stop manually, this is how the Web works.